java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java

```diff
@@ -679,12 +689,14 @@ public void switch_statement_with_expression_in_case() {
       "void fun() { int a; switch(b) { case c : System.out.println(1);break; case d || e: System.out.println(2);break;} }");
```

Comment:
even if syntactically valid, this won't compile. the boolean OR operator can not be used here, as the switch can not be used with a boolean as parameter. (also out of scope of the ticket)